lib/airbrake-ruby/notifier.rb

```diff
@@ -49,14 +49,14 @@ def initialize(user_config)
     ##
     # @macro see_public_api_method
     def notify(exception, params = {})
-      send_notice(exception, params)
-      nil
+      send_notice(exception, params, default_sender)
     end
 
     ##
     # @macro see_public_api_method
     def notify_sync(exception, params = {})
-      send_notice(exception, params, @sync_sender)
+      promise = send_notice(exception, params, @sync_sender)
+      promise.value || promise.reason
```

Comment:
It looks strange that you return notice id or error, but probably this is common enough in Ruby? Not sure if we can, but I would change it to always be hash with 2 fields:
- id - notice id on success
- error - error message on failure